src/scenic/core/utils.py

```diff
@@ -61,6 +71,38 @@ def alarm(seconds, handler=None, noNesting=False):
         signal.alarm(0)
         signal.signal(signal.SIGALRM, signal.SIG_DFL)
 
+def loadMesh(path, filetype=None, compressed=None):
+    working_path = path
+
+    # Check if file is compressed
+    if compressed is None:
+        root, ext = os.path.splitext(working_path)
+
+        if ext == ".bz2":
+            compressed = True
+            working_path = root
+        else:
+            compressed = False
+
+    # Check mesh filetype
+    if filetype is None:
+        root, ext = os.path.splitext(working_path)
+
+        if ext == "":
+            raise ValueError("Mesh filetype not provided, but could not be extracted")
+
+        filetype = ext
+
+    if compressed:
+        open_function = bz2.open
+    else:
+        open_function = open
+
+    with open_function(path, "r") as mesh_file:
```

Comment:
Does file mode "r" work for all supported formats? You might need "rb" for binary formats.

Reply:
Added an option to open files as binary files.

Reply:
But the user should not need to know about that: e.g. if they just have a file they got online and don't know how the format works. I had a look at how `trimesh.load` handles this, and if I understand correctly they always open the file in binary mode and decode as needed. So maybe you can just always open in binary mode. Also it looks like they handle `.tar.bz2` files, so you should check if that works for single meshes -- if so you may not need to do anything in this function beyond calling `trimesh.load`.